### ISSUE TITLE
update local version of containing folder when a new items comes in from remote

### ISSUE THREAD

**Issue by michielbdejong** [remotestorage/remotestorage.js#759]
When a folder listing is fetched, and we become aware of items that exist remotely but not locally, the entry in `node.local.itemsMap` is set to `false`. This is good because it means when you do getListing, you only see items that actually exist in the cache.

However, when such an item is then later fetched, its entry in the containing folder's `node.local.itemsMap` should be changed from `false` to `true`, and we're not currently doing that.

**Comment by skddc** [comment]
This is marked as blocker, but I think nobody currently understands the actual implications of the wrong behaviour, and yet we really want to release a new stable version with all the bugfixes since 0.11.0. @michielbdejong Do you think we should not release without fixing this? Or can we keep the blocker label, but at least release 0.11.1 to get all the other fixes out?

**Comment by michielbdejong** [comment]
Yes, congrats on the 0.11.1 release! Ping me when you or someone else has time to pair on this one.

**Comment by skddc** [comment]
Unfortunately, our Internet connection here on Dominica is not sufficient for remote pairing.

**Comment by silverbucket** [comment]
Hey @michielbdejong, I thought I remembered that, on our last call, you said you'd make some unit tests to illustrate the issue, do you think you'll have some time to do that soon?

**Comment by skddc** [comment]
Yer, that would explain it to the person who wants to fix it then.

**Comment by michielbdejong** [comment]
Yes, but we'll do that as part of the pairing session. And then immediately fix it while we're at it. Let me know as soon as someone has time.

**Comment by skddc** [comment]
If you just write a test, someone can pick it up and fix it without pairing. As I said, Internet connection here is not sufficient for remote pairing.

**Comment by michielbdejong** [comment]
@skddc do you want to pair on this now?

**Comment by lewisl9029** [comment]
I think I'm seeing this issue now.

After connecting a remotestorage account, I can see IndexedDB has already been populated with all the data in the module, but when I call privateClient.getAll('', false), the promise resolves with a map that's missing parts of the data, and it seems the missing data matches up perfectly with the keys in the itemsMap that are marked as false.

Seems to be reproducible with the following procedure:
1) Store some data locally in a remoteStorage module without connecting an account
2) Connect a remoteStorage account that has existing data in the same module as used in step 1
3) Wait for sync to finish and try calling privateClient.getAll('', false)
4) Try to find the missing data (populating the module with a large number of sequential keys in step 2 should make missing ones easy to find)

Any ideas of possible workarounds I could use in the mean time until this is fixed?

**Comment by skddc** [comment]
I can pair on this now.
